tools/cell_census_builder/validate.py

```diff
@@ -288,6 +302,35 @@ def count_nonzero(arr: Union[sparse.spmatrix, npt.NDArray[Any]]) -> int:
     return True
 
 
+def _validate_X_layer_has_unique_coords(args: Tuple[str, ExperimentBuilder, str, int, int]) -> bool:
+    """Validate that all X layers have no duplicate coordinates"""
+    soma_path, experiment_builder, layer_name, row_range_start, row_range_stop = args
+    census = soma.Collection(soma_path, context=SOMA_TileDB_Context())
+    census_data = census[CENSUS_DATA_NAME]
+    se = census_data[experiment_builder.name]
+    logging.info(
+        f"validate_no_dups_X start, {experiment_builder.name}, {layer_name}, rows [{row_range_start}, {row_range_stop})"
+    )
+    if layer_name not in se.ms["RNA"].X:
+        return True
+
+    X_layer = se.ms["RNA"].X[layer_name]
+    n_rows, n_cols = X_layer.shape
+    ROW_SLICE_SIZE = 100_000
+
+    for row in range(row_range_start, min(row_range_stop, n_rows), ROW_SLICE_SIZE):
+        # work around TileDB-SOMA bug #900 which errors if we slice beyond end of shape
+        end_row = min(row + ROW_SLICE_SIZE, X_layer.shape[0] - 1)
+        slice_of_X = X_layer.read(coords=(slice(row, end_row),)).tables().concat()
```

Comment:
just to visually group the comment with just the code it pertains to
```suggestion

        slice_of_X = X_layer.read(coords=(slice(row, end_row),)).tables().concat()
```